checked_message.go

```diff
@@ -20,16 +20,24 @@
 
 package zap
 
-import "github.com/uber-go/atomic"
+import "sync"
+
+var _cmPool = sync.Pool{
+	New: func() interface{} {
+		return &CheckedMessage{
+			lvl: invalidLevel,
+		}
+	},
+}
 
 // A CheckedMessage is the result of a call to Logger.Check, which allows
 // especially performance-sensitive applications to avoid allocations for disabled
 // or heavily sampled log levels.
 type CheckedMessage struct {
-	logger Logger
-	used   atomic.Uint32
-	lvl    Level
-	msg    string
+	safeToWrite bool
```

Comment:
why reorder the fields? if `safeToWrite` is next to `Level`, the struct should take up less space.